File: Diagrammatize.Design.Pattern/code/cxx/Interpreter/Sample/PrimitiveCommandNode.cpp
#include "PrimitiveCommandNode.hpp"
#include "ParseException.hpp"

PrimitiveCommandNode::~PrimitiveCommandNode() {}

void PrimitiveCommandNode::parse(Context& context) {
    name_ = context.currentToken();
    context.skipToken(name_);
    if (name_ != "go" && name_ != "right" && name_ != "left") {
        throw ParseException(name_ + " is undefined");
    }
}

std::string PrimitiveCommandNode::toString() {
    return name_;
}

File: Diagrammatize.Design.Patterns/code/python/Interpreter/Sample/PrimitiveCommandNode.py

from Node import Node
from ParseException import ParseException

# <primitive command> ::= go | right | left
class PrimitiveCommandNode(Node):
    def __init__(self):
        super().__init__()
        self.name = ''

    def parse(self, context):
        self.name = context.currentToken()
        context.skipToken(self.name)
        if not self.name in ["go", "right", "left"]:
            raise ParseException("{} is undefined".format(self.name))

    def __str__(self):
        return self.name
